tests/operations/comparison/Equal.cpp

```diff
@@ -1,11 +1,71 @@
 #define CATCH_CONFIG_MAIN
 #include "Catch/single_include/catch2/catch.hpp"
 
+#include "sgpl/algorithm/execute_core.hpp"
+#include "sgpl/hardware/Core.hpp"
 #include "sgpl/operations/comparison/Equal.hpp"
+#include "sgpl/program/Program.hpp"
+#include "sgpl/spec/Spec.hpp"
+
+// define libray and spec
+using library_t = sgpl::OpLibrary<sgpl::Equal>;
+using spec_t = sgpl::Spec<library_t>;
+
+TEST_CASE("Test Not Equal") {
```

Comment:
Ensure test cases are labeled Test ClassName, description

For example, 

Test Equal, operands not equal

Reply:
Fix elsewhere